molgenis-one-click-importer/src/main/java/org/molgenis/oneclickimporter/job/OneClickImportJob.java

```diff
@@ -27,53 +37,77 @@
 
-		String fileExtension = filename.substring(filename.lastIndexOf('.') + 1);
-		String dataCollectionName = filename.substring(0, filename.lastIndexOf('.'));
-
-		DataCollection dataCollection;
-		if (fileExtension.equals("xls") || fileExtension.equals("xlsx"))
+		List<DataCollection> dataCollections = newArrayList();
+		if (fileExtension == null)
+		{
+			throw new UnknownFileTypeException(
+					String.format("File [%s] does not have a valid extension, supported: [csv, xlsx, zip, xls]",
+							filename));
+		}
+		else if (fileExtension.equals("xls") || fileExtension.equals("xlsx"))
 		{
-			progress.progress(1, "Creating sheet from Excel");
-			Sheet sheet = excelService.buildExcelSheetFromFile(file);
-			progress.progress(2, "Creating dataCollection");
-			dataCollection = oneClickImporterService.buildDataCollection(dataCollectionName, sheet);
+			List<Sheet> sheets = excelService.buildExcelSheetsFromFile(file);
+			dataCollections.addAll(oneClickImporterService.buildDataCollectionsFromExcel(sheets));
 		}
 		else if (fileExtension.equals("csv"))
 		{
-			progress.progress(1, "Reading lines from CSV");
 			List<String> lines = csvService.buildLinesFromFile(file);
-			progress.progress(2, "Creating dataCollection");
-			dataCollection = oneClickImporterService.buildDataCollection(dataCollectionName, lines);
+			dataCollections.add(
+					oneClickImporterService.buildDataCollectionFromCsv(createValidNameFromFileName(filename), lines));
 		}
-		else
+		else if (fileExtension.equals("zip"))
 		{
-			throw new UnknownFileTypeException(
-					String.format("File with extension: %s is not a valid one-click importer file", fileExtension));
+			List<File> filesInZip = unzip(file);
+			for (File fileInZip : filesInZip)
+			{
+				String fileInZipExtension = findExtensionFromPossibilities(fileInZip.getName(), newHashSet("csv"));
+				if (fileInZipExtension != null)
+				{
+					List<String> lines = csvService.buildLinesFromFile(fileInZip);
+					dataCollections.add(oneClickImporterService.buildDataCollectionFromCsv(
+							createValidNameFromFileName(fileInZip.getName()), lines));
+				}
+				else
+				{
+					throw new UnknownFileTypeException("Zip file contains files which are not of type CSV");
+				}
+			}
 		}
 
-		progress.progress(3, "Inserting table into database");
-		EntityType entityType = entityService.createEntityType(dataCollection);
-		progress.progress(4, "Table created");
+		List<EntityType> entityTypes = newArrayList();
+		String packageName = createValidNameFromFileName(filename);
+		dataCollections.forEach(
+				dataCollection -> entityTypes.add(entityService.createEntityType(dataCollection, packageName)));
+
+		permissionSystemService.giveUserWriteMetaPermissions(entityTypes);
+		return entityTypes;
+	}
 
-		return entityType;
+	private String createValidNameFromFileName(String filename)
+	{
+		String packageName = getFileNameWithoutExtension(filename);
+		return packageName.replaceAll("[^a-zA-Z0-9_#]+", "_");
```

Comment:
please add comment explaining that regex is supposed to be doing, consider moving the regex to a named function and add unit test, even better consider rewriting regex using String class